frontend/ui/widget/sortwidget.lua

```diff
@@ -611,6 +608,35 @@ function SortWidget:onCancel()
     return true
 end
 
+function SortWidget:onHoldNonTouch()
+    -- Handle keyboard-triggered hold events by acting on the focused item directly
+    if not self.selected or not self.selected.y then
+        return true
+    end
+
+    local focused_row = self.selected.y
+    local idx_offset = (self.show_page - 1) * self.items_per_page
+    local page_last = math.min(idx_offset + self.items_per_page, #self.item_table)
+    local items_on_page = page_last - idx_offset
+
+    -- Skip if focus is on footer or invalid
+    if focused_row < 1 or focused_row > items_on_page then
+        return true
+    end
+
+    local item_idx = idx_offset + focused_row
+    local item = self.item_table[item_idx]
```

Comment:
Can this be used instead?
https://github.com/koreader/koreader/blob/672be80e4946b646761f305c772f1e5a3302360a/frontend/ui/widget/focusmanager.lua#L431